### DISCUSSION TITLE
Tired of not getting the ssl work

### DISCUSSION THREAD

**Discussion by iahadaman** [AzuraCast/AzuraCast#4624] [Q&A] [answered]
Hey Everyone!

For a couple of days, I have been trying to install SSL on my azuracast that is installed using Docker in an Azure VM Ubuntu 18.

AzuraCast is being installed very easily but the letsencrypt is not working whatsoever.

I installed letsencrypt - point my subdomain to the IP address. 
Ran the letsencrypt and added the subdomain link.

It says congratulations but doesn't works.

I opened docker-compose.yml and it shows the path for letsencrypt to be '/etc/nginx/certs'.

But **/etc/nginx/certs'** doesn't exists as well.

**Comment by ghost** [comment, marked as answer]
Hi,

There is an easy fix for this when you are running Azuracast inside a Docker Container, just install nginx proxy manager as a container and you can very easily issue SSL Certificates to Azuracast or any other Conatainer for that matter.

Kind Regards,

**Reply by BusterNeece** [reply]
@naudes Just as a heads-up, now that I'm going back and reading this...we actually have an nginx proxy manager that we ourselves include in our setup, and that's what handles our LetsEncrypt setup for us. There's no reason our built-in LetsEncrypt setup wouldn't be sufficient for these purposes, so I wouldn't recommend for most users that they go through the process of installing _yet another_ proxy layer on top of that, as that could no doubt introduce issues relating to performance, latency and our ability to track/identify listeners properly.

**Comment by iahadaman** [comment]
Hey @naudes - Thank you so much for replying.

Is it possible for you to drop any sort of tutorial? That would really helpful.

Regards.

**Comment by ghost** [comment]
Hi,
 There is a guy on YouTube his channel is called "opensource is awesome" he
has the best videos on this, if I am not mistaken he has a video he made a
week ago exactly on this topic. There is also the official nginx proxy
manager website, it describes in detail how to install it as a portainer
container.

Kind Regards,

**Reply by ghost** [reply]
I can't find the opensource is awesome channel can you give us the path?

**Comment by ghost** [comment]
Hi,

I found this and it works, I just installed it in a few minutes on a
clean Ubunto 20.04 installation.
https://shownotes.opensourceisawesome.com/install-4-apps-in-under-5-min/

**Reply by ghost** [reply]
are there any tutorial videos for Ansible install? I haven't found docs or forum post that help me
Powered by AzuraCast • v0.16.1 Stable • Ansible • PHP 8.1

**Comment by iahadaman** [comment]
Hey Naudes, I was able to install Nginx Proxy Manager and  I was able to add my subdomain to the SSL Certificates. My sub domain is pointed to the server with A record and is working fine.

but even after adding my sub domain to the SSL certificate in the Nginx Proxy Manager, when I go to the https, it gives me SSL error.

Any thoughts?

**Comment by ghost** [comment]
Ali,

Can you please send me a screenshot of your proxy manager setup for that
sub domain.

**Comment by iahadaman** [comment]
@naudes 
![11](https://user-images.githubusercontent.com/73393210/135456592-6ce21690-a43e-4a20-9af6-4e85e1fc25bf.PNG)
![12](https://user-images.githubusercontent.com/73393210/135456612-e4f2966f-fce3-4409-9603-01caaca85247.PNG)

**Comment by iahadaman** [comment]
Hello, the problem is fixed using the Nginx proxy manager <3 Thank you so much mate!

**Comment by ghost** [comment]
SlvrEagle23,

I was only trying in a simple way for him to get past his SSL problem
without getting too technical, in forums like these it is absolutely
unclear what the skills of the person is who are asking the question,
I was trying to help this guy without getting too deeply involved.

It seemed to have fixed his problem.
Kind Regards,
